core/src/structures/account_data.rs

```diff
@@ -7,7 +9,7 @@ use crate::commitment_utils::Commitment;
 #[derive(Clone, Eq, PartialEq, Debug)]
```

Comment:
how is equality defined here? by pubkey or pubkey+slot? 

what about the **Arc**?

> Two `Arc`s are equal if their inner values are equal, even if they are stored in different allocation.

Reply:
Yes you are right